libkbfs/fetch_decider.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libkbfs
+
+import (
+	"sync"
+	"time"
+
+	"github.com/keybase/client/go/logger"
+	"golang.org/x/net/context"
+)
+
+// fetchDecider is a struct that helps avoid having too frequent calls
+// into a remote server.
+type fetchDecider struct {
+	log     logger.Logger
+	fetcher func(ctx context.Context) error
+	tagKey  interface{}
+	tagName string
+	clock   clockGetter
```

Comment:
Is there a reason this can't just be an embedded struct, allowing us to do `fd.Clock()` rather than `fd.clock.Clock()`?